src/NUnitFramework/framework/Constraints/CollectionTally.cs

```diff
@@ -28,22 +28,22 @@ namespace NUnit.Framework.Constraints
 {
     /// <summary><see cref="CollectionTally"/> counts (tallies) the number of occurrences 
     /// of each object in one or more enumerations.</summary>
-    public class CollectionTally
+    public sealed class CollectionTally
     {
         /// <summary>The result of a <see cref="CollectionTally"/>.</summary>
-        public class CollectionTallyResult
+        public sealed class CollectionTallyResult
         {
             /// <summary>Items that were not in the expected collection.</summary>
-            public List<object> ExtraItems { get; set; }
+            public List<object> ExtraItems { get; }
 
             /// <summary>Items that were not accounted for in the expected collection.</summary>
-            public List<object> MissingItems { get; set; }
+            public List<object> MissingItems { get; }
 
             /// <summary>Constructs an empty <see cref="CollectionTallyResult"/>.</summary>
-            public CollectionTallyResult()
+            public CollectionTallyResult(List<object> missingItems, List<object> extraItems)
             {
-                ExtraItems = new List<object>();
-                MissingItems = new List<object>();
+                MissingItems = missingItems;
+                ExtraItems = extraItems;
```

Comment:
Why this change? And if this change, does the doccoment need updating too - as surely this allows you to potentially create a non-empty CollectionTallyResult

Reply:
It allowed
```cs
return new CollectionTallyResult()
{
    MissingItems = new List<object>(_missingItems),
    ExtraItems = new List<object>(_extraItems)
};
```
to become:
```cs
return new CollectionTallyResult(
    new List<object>(_missingItems),
    new List<object>(_extraItems));
```
rather than:
```cs
var temp = new CollectionTallyResult();
temp.MissingItems.AddRange(_missingItems);
temp.ExtraItems.AddRange(_extraItems);
return temp;
```

Thanks for catching the doc comment! What do you think, is this too far? Should we go back to allowing the list instances (themselves mutable) to be swapped out, even though we only used that for initialization? What would it mean to later on set them to null or a new list, as opposed to clearing them?

Reply:
I understand, thanks. Sounds sensible - maybe just the comment to change. 😄